### PR TITLE
look for dmi table also in sysfs (bsc #1010276)

### PR DESCRIPTION
**Pull request by wfeldt**

### CODE REVIEW

File: src/hd/hd.c [hunk 1]
```diff
@@ -6067,16 +6067,26 @@ char *get_sysfs_attr_by_path(const char *path, const char *attr)
...
-  static char buf[1024];
-  int i, fd;
+  static char *buf = NULL;
+  char * ptr;
+  int i, fd, max;
 
   if(len) *len = 0;
 
+  // init static buffer on first run
+  if(!buf) buf = new_mem(MAX_ATTR_SIZE + 1);
```
Comment (lslezak) [inline src/hd/hd.c]: I cannot see any deallocation, is that OK? I mean consuming 64kB RAM is not that bad these days, but if we can easily free that memory at some point we should do that...

Comment (wfeldt) [inline src/hd/hd.c]: it's a static buffer; it's used all the time; you would free it on exit() which would be pointless

Comment (lslezak) [inline src/hd/hd.c]: OK, but what if `libhd` is used from another application like YaST? Would that make sense?

Comment (lslezak) [inline src/hd/hd.c]: No, but after finishing the hardware detection we could deallocate the not needed memory. But if the architecture does not allow doing it easily I'm fine with the current state.

Comment (wfeldt) [inline src/hd/hd.c]: I see. It was a static array before, I wouldn't bother too much; there are several places in libhd where static data structures are allocated.

Comment (lslezak) [inline src/hd/hd.c]: OK, let's keep it as it is...

**Pull request merged**